src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchRestTemplate.java

```diff
@@ -339,26 +282,27 @@ public SearchResponse suggest(SuggestBuilder suggestion, IndexCoordinates index)
 		SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
 		sourceBuilder.suggest(suggestion);
 		searchRequest.source(sourceBuilder);
-
-		try {
-			return client.search(searchRequest, RequestOptions.DEFAULT);
-		} catch (IOException e) {
-			throw new ElasticsearchException("Could not execute search request : " + searchRequest.toString(), e);
-		}
-
+		return execute(client -> client.search(searchRequest, RequestOptions.DEFAULT));
 	}
 
 	@Override
 	protected MultiSearchResponse.Item[] getMultiSearchResult(MultiSearchRequest request) {
-		MultiSearchResponse response;
-		try {
-			response = client.multiSearch(request, RequestOptions.DEFAULT);
-		} catch (IOException e) {
-			throw new ElasticsearchException("Error for search request: " + request.toString(), e);
-		}
+		MultiSearchResponse response = execute(client -> client.multiSearch(request, RequestOptions.DEFAULT));
 		MultiSearchResponse.Item[] items = response.getResponses();
 		Assert.isTrue(items.length == request.requests().size(), "Response should has same length with queries");
 		return items;
 	}
 	// endregion
+
+	@Override
+	public <T> T execute(RestHighLevelClientCallback<T> callback) throws RuntimeException {
+
+		Assert.notNull(callback, "callback must not be null");
+
+		try {
+			return callback.doWithClient(client);
+		} catch (Throwable t) {
```

Comment:
Limit the caught exceptions to the actual ones such as `RuntimeException` and `IOException`.

Reply:
that would mean that the `ClientCallback.doWithClient()` method could only throw IOException as a checked one. Can we put this limitation on that callback method?

Reply:
Yes, that change would indeed make a lot of sense. Implementors aren't required to throw exceptions so we should be good to extend the list of potentially thrown exceptions if we recognize the need for it.